server/src/main/java/com/cloud/consoleproxy/ConsoleProxyManagerImpl.java

```diff
@@ -1006,6 +1006,13 @@ private void allocCapacity(long dataCenterId) {
     }
 
     public boolean isZoneReady(Map<Long, ZoneHostInfo> zoneHostInfoMap, long dataCenterId) {
+        List <HostVO> hosts = _hostDao.listByDataCenterId(dataCenterId);
+        if (CollectionUtils.isEmpty(hosts)) {
+            if (s_logger.isDebugEnabled()) {
+                s_logger.debug("Zone " + dataCenterId + " has no host available which is enabled and in Up state");
+            }
+            return false;
+        }
```

Comment:
this is the exact same code as inSecondaryStorageManagerImpl.isZoneReady() can you factor out and reuse?